course_gather/viewsets.py

```diff
@@ -52,7 +52,7 @@ class CourseViewSet(viewsets.ModelViewSet):
     filter_backends = (filters.DjangoFilterBackend,)
 
     # Gzip, then cache for 5 minutes
-    @method_decorator([gzip_page, cache_page(60 * 5)])
+    @method_decorator(cache_page(60 * 5))
```

Comment:
You should zip more than just the course view. This is the problem child currently, but all viewsets will benefit from a zipped response.

You may need to set the Content-Encoding header and the Accept-Encoding header in the front end as well.

Reply:
Sorry, not understanding what you're saying. All API responses are now gzip'ed.

The accept-encoding header is set automatically by the browser.

Reply:
Yes but the browser isn't making the request to the API backend, React is. What I meant is that I'm not entirely certain that the React request will have all of the needed headers.

Reply:
I just tested and it seems to be working. I'm not sure how it's accomplishing that but it's good that it works.

Reply:
I just reread your post and realized I messed up.

I meant we should be caching all responses for the entire site.
Sorry about the confusion.

Reply:
> I'm not sure how it's accomplishing that but it's good that it works.

[The browser handles everything for us](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation).

> I meant we should be caching all responses for the entire site.

I think for now it makes more sense to cache on a per API route basis. For example, we may want to cache transfer courses longer than we cache regular courses for.

Reply:
I disagree totally with this. I think there's no good reason **not** to cache all endpoint responses.

Reply:
I'm not against caching everything. I'm just saying that it would be nice to be able to specify different cache times for different views / API routes. If there's a cleaner way to do that then what's here right now, I'm all for that.

Reply:
I don't really understand the why of different cache I guess. Can you explain why?

Reply:
I'm confused in what situation we'd have that would require different levels of caching.  As of right now we only have data scraped off of banweb (which only updates once a week at a maximum really).  Even if we do a later integration with rate my professor type stuff, those listings generally only get rating spikes at around the end of semesters.  Same spikes tend to happen for transfer stuff as well (majority anyways).

Where does a difference in caching actually make sense?  Realistically, our cache's TTL could be damn near whatever within a few days time and it'd make no difference realistically.

What would be a case where we would need different cache times for different routes?

Reply:
Banweb course listings (if we start scraping those) I assume would be scrapped much more frequently at an interval of something like 30 - 60 minutes, if for nothing else to update the number of available seats around registration time.

I guess just caching everything for 15 minutes could work. 🤷

Reply:
I went ahead and added settings to prod to cache everything for 15 minutes: 9a6c365. Let me know if that works.